### ISSUE TITLE
Support DBT `modules` in jinja templating

### ISSUE THREAD

**Issue by mroy-seedbox** [sqlfluff/sqlfluff#6470]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

DBT exposes a [modules variable](https://docs.getdbt.com/reference/dbt-jinja-functions/modules) which can be used to access some python modules: `datetime`, `pytz`, `re`, and `itertools`.

- The first error we run into is `Undefined jinja template variable: 'modules'`.
- Then if we define a template variable: `modules = {"datetime": {"date": {}}}`, we get this error when we try to use `modules.datetime.something()`: `Unrecoverable failure in Jinja templating: 'dict' object is not callable.`

Is there any way to define a template variable as a callable?

### Dialect

Any.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mroy-seedbox** [comment]
I think it should be doable for SQLFluff to expose these python modules as well.

But in the meantime, a workaround is to wrap the methods that we need in a macro, and them configure the macro in `[sqlfluff:templater:jinja:macros]`.

**Comment by alanmcruickshank** [comment]
Sounds sensible - and I think you might be the best person to implement it too!

**Comment by mroy-seedbox** [comment]
Related: #6564

**Comment by ryanbordo** [comment]
Wondering if there is any progress here, I am also having this issue and willing to contribute a fix.

**Comment by mroy-seedbox** [comment]
I haven't had any time to work on it.

And in the meantime we're using the workaround of wrapping the module calls in macros.

**Comment by peterbud** [comment]
@mroy-seedbox - is this issue still relevant? Or can we close it?

**Comment by mroy-seedbox** [comment]
Yeah, it is still relevant. But at least there's a simple workaround in the meantime.
